VulkanLaunchpad.cpp

```diff
@@ -699,12 +700,34 @@ VkDeviceMemory vklAllocateHostCoherentMemoryForGivenRequirements(VkDeviceSize bu
 	}
 }
 
+VkDeviceMemory vklAllocateDeviceMemoryForGivenRequirements(VkDeviceSize bufferSize, VkMemoryRequirements memoryRequirements)
+{
+	const auto memoryAllocInfo = vklCreateMemoryAllocateInfo(bufferSize, memoryRequirements, VK_MEMORY_PROPERTY_DEVICE_LOCAL_BIT);
+
+	// Allocate:
+	VkDeviceMemory memory;
+	VkResult returnCode = vkAllocateMemory(static_cast<VkDevice>(mDevice), &memoryAllocInfo, NULL, &memory);
+	if (returnCode == VK_SUCCESS) {
+		return memory;
+	}
+	else {
+		VKL_EXIT_WITH_ERROR(std::string("Error allocating memory of size [") + std::to_string(bufferSize) + "] and requirements[" + std::to_string(memoryRequirements.alignment) + ", " + std::to_string(memoryRequirements.memoryTypeBits) + ", " + std::to_string(memoryRequirements.size) + "]\n    Error Code: " + to_string(returnCode));
+	}
+}
+
```

Comment:
This is _exactly_ the same function as `vklAllocateHostCoherentMemoryForGivenRequirements` except for the different memory property flags. 
Let's avoid code duplication and combine
- `VkDeviceMemory vklAllocateHostCoherentMemoryForGivenRequirements(VkDeviceSize bufferSize, VkMemoryRequirements memoryRequirements)` and
- `VkDeviceMemory vklAllocateDeviceMemoryForGivenRequirements(VkDeviceSize bufferSize, VkMemoryRequirements memoryRequirements)` 

into:
- `VkDeviceMemory vklAllocateDeviceMemoryForGivenRequirements(VkDeviceSize bufferSize, VkMemoryRequirements memoryRequirements, VkMemoryPropertyFlags flags)` !

Reply:
I know, but I wasn't sure if you wanted to keep the different function names to make it easier

Reply:
Yeah, but this is actually only used internally anyways, right? So, for a function which is only used internally, doesn't hurt to add another parameter.